ompi/mca/osc/rdma/osc_rdma_component.c

```diff
@@ -726,16 +737,20 @@ static int allocate_state_shared (ompi_osc_rdma_module_t *module, void **base, s
 static int ompi_osc_rdma_query_mtls (void)
 {
     char **mtls_to_use;
+    bool mtl_match = false;
 
     mtls_to_use = opal_argv_split (ompi_osc_rdma_mtl_names, ',');
     if (mtls_to_use && ompi_mtl_base_selected_component) {
-	for (int i = 0 ; mtls_to_use[i] ; ++i) {
-	    if (0 == strcmp (mtls_to_use[i], ompi_mtl_base_selected_component->mtl_version.mca_component_name)) {
-		return OMPI_SUCCESS;
-	    }
-	}
+        for (int i = 0 ; mtls_to_use[i] ; ++i) {
+            if (0 == strcmp (mtls_to_use[i], ompi_mtl_base_selected_component->mtl_version.mca_component_name)) {
+                mtl_match = true;
+                break;
+            }
+        }
     }
-    return -1;
+
+    opal_argv_free (mtls_to_use);
+    return mtl_match ? OMPI_SUCCESS : -1;
```

Comment:
Should probably return `OMPI_ERR_NOT_FOUND` or somesuch; not just a straight `-1`.  That way the caller can still check for `OMPI_SUCCESS` or not -- they don't necessarily have to look for a specific return code, but they *could* still look for `<0` if they wanted to check for an error.